app/modules/cdr/endpoints_cdr.py

```diff
@@ -351,6 +355,197 @@ async def get_online_sellers(
+            )
+    for column in df.columns:
+        df[column] = df[column].apply(lambda x: x if pd.notna(x) else "")
+    return df
+
+
+@module.router.post(
+    "/cdr/sellers/{seller_id}/results/",
+    status_code=200,
+)
+async def send_seller_results(
+    seller_id: UUID,
+    email: schemas_cdr.ResultRequest,
+    db: AsyncSession = Depends(get_db),
+    user: models_core.CoreUser = Depends(is_user_a_member_of(GroupType.admin_cdr)),
+):
+    """
+    Get a seller's results.
+
+    **User must be CDR Admin to use this endpoint**
+    """
+    seller = await cruds_cdr.get_seller_by_id(db, seller_id)
+    if not seller:
+        raise HTTPException(
+            status_code=404,
+            detail="Seller not found.",
+        )
+    seller_group = await cruds_groups.get_group_by_id(db, seller.group_id)
+    if seller_group is None:
+        raise HTTPException(
+            status_code=404,
+            detail="Seller group not found.",
+        )
+    products = await cruds_cdr.get_products_by_seller_id(db, seller_id)
+    if len(products) == 0:
+        raise HTTPException(
+            status_code=400,
+            detail="There is no products for this seller so there is no results to send.",
+        )
+    variants: list[models_cdr.ProductVariant] = []
+    product_fields: dict[UUID, list[models_cdr.CustomDataField]] = {}
+    for product in products:
+        product_variants = await cruds_cdr.get_product_variants(db, product.id)
+        variants.extend(product_variants)
+        product_fields[product.id] = list(
+            await cruds_cdr.get_product_customdata_fields(
+                db,
+                product.id,
+            ),
+        )
+    variant_ids = [v.id for v in product_variants]
+    purchases = await cruds_cdr.get_all_purchases(db)
+    if len(purchases) == 0:
+        raise HTTPException(
+            status_code=400,
+            detail="There is no purchases in the database so there is no results to send.",
+        )
+    purchases_by_users: dict[str, list[models_cdr.Purchase]] = {}
+    users = await get_users(db)
+    for purchase in purchases:
+        if purchase.product_variant_id in variant_ids:
+            if purchase.user_id not in purchases_by_users:
+                purchases_by_users[purchase.user_id] = []
+            purchases_by_users[purchase.user_id].append(purchase)
+    users_answers = {}
+    for each_user in users:
+        users_answers[user.id] = list(
+            await cruds_cdr.get_customdata_by_user_id(
+                db,
+                each_user.id,
+            ),
+        )
+    df = construct_dataframe_from_users_purchases(
+        users_purchases=purchases_by_users,
+        users=list(users),
+        products=list(products),
+        variants=variants,
+        data_fields=product_fields,
+        users_answers=users_answers,
+    )
+    file_path = f"/tmp/CdR {datetime.now(tz=UTC).year} ventes {seller.name}.xlsx"  # noqa: S108
```

Comment:
I'm not completely sure if it would work, but S108 looks interesting.

Reply:
If we don't use a temp file as proposed by S108, please place this file in `/data/...` which is gitignored.
Always use file manipulation utils to access the file system, to prevent injections in filepath

Reply:
The file creation is done by pandas it does not seem to be compatible